### PR TITLE
Update documentation to use doctests

### PR DESCRIPTION
**Pull request by omus**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@
 *.jl.mem
 /docs/build/
 /docs/site/
+Manifest.toml
```
Comment (ararslan) [inline .gitignore]: Perhaps add `docs/Manifest.toml` as well?

Comment (omus) [reply .gitignore]: This rule ignores all files named `Manifest.toml`. `/Manifest.toml` would be how to ignore only the root file.

Comment (ararslan) [reply .gitignore]: Oh whoa, TIL!

**Comment by omus** [comment]
> Approved assuming the resulting docs look good

The docs could not look any better ;)

**Pull request merged**